diff --git a/crates/c-api/include/wasi.h b/crates/c-api/include/wasi.h
index 92271496c..35a29069f 100644
--- a/crates/c-api/include/wasi.h
+++ b/crates/c-api/include/wasi.h
@@ -7,7 +7,7 @@
 #ifndef WASI_H
 #define WASI_H
 
-#include "wasm.h"
+#include "wasmtime/include/wasm.h"
 
 #ifndef WASI_API_EXTERN
 #ifdef _WIN32
diff --git a/crates/c-api/include/wasmtime.h b/crates/c-api/include/wasmtime.h
index 75880ce02..21bb3c6f0 100644
--- a/crates/c-api/include/wasmtime.h
+++ b/crates/c-api/include/wasmtime.h
@@ -12,8 +12,8 @@
 #ifndef WASMTIME_API_H
 #define WASMTIME_API_H
 
-#include <wasm.h>
-#include <wasi.h>
+#include "wasmtime/include/wasm.h"
+#include "wasmtime/crates/c-api/include/wasi.h"
 
 #ifdef __cplusplus
 extern "C" {
