# Fix linking on Windows.

diff --git a/bazel/defs.bzl b/bazel/defs.bzl
index 0e9559a65f..c289f3e866 100644
--- a/bazel/defs.bzl
+++ b/bazel/defs.bzl
@@ -137,6 +137,7 @@ def _default_args():
             ],
             "@v8//bazel/config:is_windows": [
                 "/std:c++17",
+                "/bigobj",
             ],
             "//conditions:default": [],
         }) + select({
@@ -155,6 +156,7 @@ def _default_args():
         includes = ["include"],
         linkopts = select({
             "@v8//bazel/config:is_windows": [
+                "/DEBUG:FULL",
                 "Winmm.lib",
                 "DbgHelp.lib",
                 "Advapi32.lib",
