# Fix linking on Windows.

diff --git a/bazel/defs.bzl b/bazel/defs.bzl
index f162149f0a..8e2db11b48 100644
--- a/bazel/defs.bzl
+++ b/bazel/defs.bzl
@@ -155,6 +155,7 @@ def _default_args():
         includes = ["include"],
         linkopts = select({
             "@v8//bazel/config:is_windows": [
+                "/DEBUG:FULL",
                 "Winmm.lib",
                 "DbgHelp.lib",
                 "Advapi32.lib",
