diff --git a/crates/cache/src/lib.rs b/crates/cache/src/lib.rs
index e1186ce28..a1bba6ea5 100644
--- a/crates/cache/src/lib.rs
+++ b/crates/cache/src/lib.rs
@@ -98,14 +98,14 @@ impl<'config> ModuleCacheEntryInner<'config> {
             format!(
                 "{comp_name}-{comp_ver}-{comp_mtime}",
                 comp_name = compiler_name,
-                comp_ver = env!("GIT_REV"),
+                comp_ver = "63d5b919306ebecc00cd39090910d89c02dcda9b",
                 comp_mtime = self_mtime,
             )
         } else {
             format!(
                 "{comp_name}-{comp_ver}",
                 comp_name = compiler_name,
-                comp_ver = env!("GIT_REV"),
+                comp_ver = "63d5b919306ebecc00cd39090910d89c02dcda9b",
             )
         };
         let root_path = cache_config.directory().join("modules").join(compiler_dir);
