--- build.rs
+++ build.rs
@@ -62,6 +62,7 @@ fn link_in_librustix_outline(arch: &str, asm_name: &str) {
         let out_dir = var("OUT_DIR").unwrap();
         Build::new().file(&asm_name).compile(&name);
         println!("cargo:rerun-if-changed={}", asm_name);
+        return;
         let from = format!("{}/lib{}.a", out_dir, name);
         let prev_metadata = std::fs::metadata(&to);
         std::fs::copy(&from, &to).unwrap();
